README.md

```diff
@@ -25,17 +56,45 @@ algorithms for signing your credentials. For now:
   a Bitcoin-based or Ethereum-based ledger) 
 * You _can_ use RSA keys to sign, if your use case requires it.
 
+#### Private Key Storage
+
+Where to store the private keys?
+
 TODO: Add a brief discussion of where to store the private keys. Point to
 several recommended Wallet or KMS libraries.
 
-## Background
+Use `await keyPair.export()`
 
-TBD
+#### Publishing the Public Key
 
-See also (related specs):
+TODO: Explain `documentLoader` / key resolvers, and where to put the public
+key so that the verifier can get to it.
 
-* [Verifiable Credentials Data Model](https://w3c.github.io/vc-data-model/)
-* [Linked Data Proofs 1.0](https://w3c-dvcg.github.io/ld-proofs/)
+#### Key ID
+
+TODO: Add discussion on typical key ID strategies
+
+* `'did:example:123' + '#' + keyPair.fingerprint()` (Ledger DID based)
+* `'did:key:' + keyPair.fingerprint()` ([`did:key` method](https://github.com/digitalbazaar/did-method-key/pull/1/files) based)
+* `https://example.com/publicKey.json`
+* `urn:123`
+
+
+#### Controller Document
+
+TODO: Explain controller document
+
+* `did:example:123` (Controller's DID on a ledger)
+* Embedded / `did:key` method
+* `https://example.com/controller.json` (published on the web)
+
+```js
+const controllerDoc = {
+  '@context': 'https://w3id.org/security/v2',
+  id: controllerId,
+  assertionMethod: [keyPair.id]
```

Comment:
I wonder if at this point in the doc that keyPair.id should not just be a URI.  Some of the examples given in the top sections of the document seem to assume that users know what a keyPair object is.  Which is only explained later in generating keys.

Reply:
@mattcollier agreed, I'm not sure what the best ordering is, since it's a circular dependency. When generating keys, you need to know about the key id (which is not always a url) and controller. (Which suggests that those should come before generating docs). 
But then to explain what a controller is, the examples sort of assume key pair. I'm hoping the variable `keyPair` will be self-explanatory enough that the example makes sense (long enough for the reader to go see how to create em)